packages/scene-composer/src/components/panels/scene-settings/GroundPlaneSettingsEditor.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useCallback, useContext, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { Button, FormField, Input, SpaceBetween } from '@awsui/components-react';
+
+import { getGlobalSettings } from '../../../common/GlobalSettings';
+import { sceneComposerIdContext } from '../../../common/sceneComposerIdContext';
+import { IGroundPlaneSettings, KnownSceneProperty, COMPOSER_FEATURES } from '../../../interfaces';
+import useLifecycleLogging from '../../../logger/react-logger/hooks/useLifecycleLogging';
+import { useStore } from '../../../store';
+import { ColorSelectorCombo } from '../scene-components/tag-style/ColorSelectorCombo/ColorSelectorCombo';
+import { parseS3BucketFromArn } from '../../../utils/pathUtils';
+
+export const GroundPlaneSettingsEditor: React.FC = () => {
+  useLifecycleLogging('GroundPlaneSettingsEditor');
+  const texturesEnabled = getGlobalSettings().featureConfig[COMPOSER_FEATURES.Textures];
+
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const intl = useIntl();
+  const setSceneProperty = useStore(sceneComposerId)(
+    (state) => state.setSceneProperty<IGroundPlaneSettings | undefined>,
+  );
+  const groundSettings = useStore(sceneComposerId)((state) =>
+    state.getSceneProperty<IGroundPlaneSettings>(KnownSceneProperty.GroundPlaneSettings),
+  );
+
+  const [internalColor, setInternalColor] = useState(groundSettings?.color || '#00FF00');
+  const [internalUri, setInternalUri] = useState(groundSettings?.textureUri || '');
+  const [internalOpacity, setInternalOpacity] = useState(groundSettings?.opacity || 0);
+
+  const showAssetBrowserCallback = useStore(sceneComposerId)(
+    (state) => state.getEditorConfig().showAssetBrowserCallback,
+  );
+
+  const groundColors = useStore(sceneComposerId)((state) =>
+    state.getSceneProperty<string[]>(KnownSceneProperty.GroundCustomColors, []),
+  );
+  const setGroundColorsSceneProperty = useStore(sceneComposerId)((state) => state.setSceneProperty<string[]>);
+
+  const onInputBlur = useCallback(() => {
+    if (groundSettings?.opacity !== internalOpacity) {
+      if (groundSettings?.textureUri) {
+        setSceneProperty(KnownSceneProperty.GroundPlaneSettings, {
+          opacity: internalOpacity,
+          textureUri: internalUri,
+        });
+      } else {
+        setSceneProperty(KnownSceneProperty.GroundPlaneSettings, {
+          opacity: internalOpacity,
+          color: internalColor,
+        });
+      }
+    }
+  }, [groundSettings, internalColor, internalOpacity, internalUri]);
+
+  const onOpacityChange = useCallback(
+    (event) => {
+      let value = Number(event.detail.value);
+      value = value / 100; //convert from percentage
+      if (value > 1) {
```

Comment:
nit-pick: You could simplify this by forcing `value` to `boolean` out the gates

``` 
const value = !!Math.max(0, Number(event.detail.value) / 100);

if (value !== internalOpacity) {
  setInternalOpacity(value);
}

Reply:
It's not a boolean it's a 0 to 1 decimal value actually that's being express as percentage in the UX and then saved in system as 0 to 1.

Reply:
Ah, then you can just eliminate the `!!` from the snippet I suggested.